src/common.c

```diff
@@ -1696,16 +1747,19 @@ static int ofi_pollfds_waitall(struct ofi_pollfds *pfds,
 		for (i = 1; i < pfds->nfds && found < ret; i++) {
 			if (pfds->fds[i].revents) {
 				events[found].events = pfds->fds[i].revents;
-				events[found++].data.ptr = pfds->ctx[i].context;
+				ctx = ofi_pollfds_get_ctx(pfds, pfds->fds[i].fd);
+				assert(ctx && ctx->context);
```

Comment:
I thought this would be a nice check, but it turns out that the socket provider sets the context to NULL in some cases.  The sockets provider uses epoll where available, but pollfds where not.  This can be hit by using configure --disable-epoll to force this path.